test/integration/agent/networkpolicy_metrics_test.go

```diff
@@ -0,0 +1,206 @@
+	networkPolicyRule1 := newPolicyRule(v1beta1.DirectionIn, from, to, services)
+	networkPolicyRule2 := newPolicyRule(v1beta1.DirectionOut, from, to, services)
+	networkPolicyRule3 := newPolicyRule(v1beta1.DirectionIn, []string{}, to, services)
+	return &v1beta1.NetworkPolicy{
+		ObjectMeta:      v1.ObjectMeta{UID: types.UID(uid), Name: uid, Namespace: testNamespace},
+		Rules:           []v1beta1.NetworkPolicyRule{networkPolicyRule1, networkPolicyRule2, networkPolicyRule3},
+		AppliedToGroups: appliedTo,
+	}
+}
+
+func TestAddMultipleNewtorkPolicy(t *testing.T) {
+	// Initialize NetworkPolicy metrics (prometheus)
+	metrics.InitializeNetworkPolicyMetrics()
+
+	controller, clientset := newTestController()
+	addressGroupWatcher := watch.NewFake()
+	appliedToGroupWatcher := watch.NewFake()
+	networkPolicyWatcher := watch.NewFake()
+	clientset.AddWatchReactor("addressgroups", k8stesting.DefaultWatchReactor(addressGroupWatcher, nil))
+	clientset.AddWatchReactor("appliedtogroups", k8stesting.DefaultWatchReactor(appliedToGroupWatcher, nil))
+	clientset.AddWatchReactor("networkpolicies", k8stesting.DefaultWatchReactor(networkPolicyWatcher, nil))
+
+	protocolTCP := v1beta1.ProtocolTCP
+	port := intstr.FromInt(80)
+	services := []v1beta1.Service{{Protocol: &protocolTCP, Port: &port}}
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	go controller.Run(stopCh)
+
+	// Test adding policy1 with a single rule
+	policy1 := newNetworkPolicy("policy1", []string{"addressGroup1"}, []string{}, []string{"appliedToGroup1"}, services)
+	networkPolicyWatcher.Add(policy1)
+	networkPolicyWatcher.Action(watch.Bookmark, nil)
+	// Wait for adding policy1 to finish.
+	if err := wait.Poll(networkPolicyInterval, networkPolicyTimeout, func() (bool, error) {
+		if controller.GetNetworkPolicy("policy1", testNamespace) == nil {
+			return false, nil
+		} else {
+			return true, nil
+		}
+	}); err != nil {
+		t.Errorf("Error: Adding networkpolicy policy1 failed: %v", err)
+	}
+	checkNetworkPolicyMetrics(t, controller)
+
+	// Test adding policy2 with multiple rules
+	policy2 := newNetworkPolicyWithMultipleRules("policy2", []string{"addressGroup2"}, []string{"addressGroup2"}, []string{"appliedToGroup2"}, services)
+	networkPolicyWatcher.Add(policy2)
+	// Wait for adding policy2 to finish.
+	if err := wait.Poll(networkPolicyInterval, networkPolicyTimeout, func() (bool, error) {
+		if controller.GetNetworkPolicy("policy2", testNamespace) == nil {
+			return false, nil
+		} else {
+			return true, nil
+		}
+	}); err != nil {
+		t.Errorf("Error: Adding networkpolicy policy2 failed: %v", err)
+	}
+	checkNetworkPolicyMetrics(t, controller)
+
+	// Test deleting policy1
+	networkPolicyWatcher.Delete(newNetworkPolicy("policy1", []string{}, []string{}, []string{}, nil))
+	// Wait for deleting policy1 to finish.
+	if err := wait.Poll(networkPolicyInterval, networkPolicyTimeout, func() (bool, error) {
```

Comment:
Usage looks good. May be it is better to keep this in a function as this was repeated multiple times for readability.

Reply:
Thank you for your suggestion. Let me put it into a function.